src/join.ts

```diff
@@ -16,18 +16,29 @@ interface SegmentBudgetState {
   takenFnSegmentLen: number
 }
 
-export const join = (
+const joinEntry = (a: any, b: any, c: any, d: any) =>
+  c != null ? joinMain(a, b, c, d) : joinCurried(a, b)
+
+export const join = joinEntry as unknown as typeof joinCurried & typeof joinMain
+
+export const joinCurried =
+  (joiner: string, segments: Transform[]) =>
+  (input: Input, options?: JoinOptions) =>
+    joinMain(input, joiner, segments, options)
```

Comment:
Without a curried form of `join`, some of the usages were just becoming very awkward and cumbersome, for example: https://github.com/snaplet/copycat/pull/32/files#diff-2d82b91fbfca95ee52a36b8f41125b2b304ecaba2f7d5d039f6837969e0c3bdeR18

`join` isn't exposed as a public api though, in case we don't want such an api in copycat